docs/source/getting_started/2_module_devguide.rst

```diff
@@ -0,0 +1,164 @@
+
+When we started Savant, we had incapable, inefficient tooling as DeepStream has. But in Savant 0.2.5, we made many improvements to give developers efficient tooling, disrupting the annoying "change, build, run, read logs" cycle. We want Savant developers to have tooling enabling building integration tests and discovering problems in seconds, not minutes.
+
+To help develop modules efficiently Savant provides provides two features and one auxiliary technology:
+
+- **Client SDK**: a Python programmatic API allowing developers to send media and meta to Savant and receive the results right from Python programs without threads, complex logic, etc;
+
+- **Dev Server**: a feature enabling pipeline reloading Python code automatically on change, significantly saving the developer's time working on Python code.
+
+The above-mentioned auxiliary technology is `OpenTelemetry <https://opentelemetry.io/>`_. It is a broad topic, but for now, you may think that it helps profiling the code performance and collecting logs and events happening in the pipeline.
+
+Let us observe basic operations you may encounter during the pipeline development and our approach to carry on them efficiently.
+
+Recommended Module Layout
+-------------------------
+
+Small and medium-size pipelines can have a flat layout with all the components placed in a single directory. You can find a number of such pipelines in `samples <https://github.com/insight-platform/Savant/tree/develop/samples>`_ directory.
+
+You don't need to have a whole Savant repo to develop a module, only what is inside a module directory.
+
+The `my-module` created from `template <https://github.com/insight-platform/Savant/tree/develop/samples/template>`_ has more universal structure suitable for advanced pipelines with dependencies. Let us discuss the structure briefly:
+
+- **.devcontainer**: the directory is required only if you are developing with VSCode; it contains Docker specification for VSCode. You may find two flavors there - for x86+dGPU and for Nvidia Jetson SOCs.
+- **docker**: the directory contains two Dockerfiles containing stub code for building custom development Docker containers, installing additional requirements from the ``requirements.txt``; it is often a way to go if the pipeline requires Python libraries not delivered with Savant base images.
+- **.dockerignore**: the file contains exclusions for files and paths typically must be avoided to add to a Docker image; you can extend them to reflect your actual needs, e.g. exclude media files.
+- **Makefile** contains commands for building and launching the Docker image in a standalone mode, without IDE.
+- **README.md** is just a README stub.
+- **requirements.txt** contains additional Python requirements to add to the custom Docker image.
+- **module** directory with actual pipeline code.
```

Comment:
I've placed the `module` dir on one level lower `src/module`, in order to mount client + module code into dev container easier.